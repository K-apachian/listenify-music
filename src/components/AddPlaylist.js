import React, { Component } from "react";
import { Form, Button, FloatingLabel, Row, Col } from "react-bootstrap";
import { connect } from "react-redux";
import { addPlaylist } from "../redux";

class AddPlaylist extends Component {
  constructor(props) {
    super(props);

    this.state = {
      playlist: "",
    };
  }

  handlePlaylistChange = (event) => {
    this.setState({
      playlist: event.target.value,
    });
  };

  handleSubmit = (event) => {
    event.preventDefault();
    const playlistData = {
      title: this.state.playlist,
      songIDs: [],
      accessCount: 0,
    };

    if (this.state.playlist === "") {
      alert(" You Forgot to Add Playlist Name");
    } else {
      alert(this.state.playlist);
      this.props.addPlaylist(playlistData);
      this.props.onHide();
    }
  };

  render() {
    return (
      <div style={{ textAlign: "center" }}>
        <Form onSubmit={this.handleSubmit}>
          <Row className="mb-3">
            <Form.Group as={Col}>
              <FloatingLabel
                controlId="floatingInput"
                label="Playlist"
                className="mb-3"
              >
                <Form.Control
                  type="text"
                  placeholder="Playlist"
                  value={this.state.playlist}
                  onChange={this.handlePlaylistChange}
                />
              </FloatingLabel>

              <Button variant="outline-dark" type="submit">
                Submit
              </Button>
            </Form.Group>
          </Row>
        </Form>
      </div>
    );
  }
}

export default connect(null, { addPlaylist })(AddPlaylist);
