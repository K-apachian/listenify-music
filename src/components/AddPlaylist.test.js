import React from "react";
import { shallow } from "enzyme";
import { expect, it } from "@jest/globals";

import AddPlaylist from "./AddPlaylist";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";

const mockStore = configureMockStore();
const store = mockStore({
  userData: {
    loading: false,
    users: [],
    user: {},
    error: "",
    isLoggedIn: false,
  },
  songsData: {
    loading: false,
    songs: [],
    error: "",
  },
  playlistData: {
    loading: false,
    playlists: [],
    error: "",
  },
});

it("should render entire AddPlaylist Component", () => {
  const wrapper = shallow(
    <Provider store={store}>
      <AddPlaylist />
    </Provider>
  );
  expect(wrapper).toMatchSnapshot();
});
