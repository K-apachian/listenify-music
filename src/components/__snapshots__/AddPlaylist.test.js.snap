// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render entire AddPlaylist Component 1`] = `ShallowWrapper {}`;
